### PR TITLE
Imply outlives-bounds on lazy type aliases

### LINKED ISSUE #118479: Imply outlives-bounds on lazy type aliases
In the most recent design meeting (2023-11-09), T-lang has decided to move forward with “treating lazy type aliases like structs” with respect to where-clauses. Concretely, this means that we want to check lazy type aliases for well-formedness without implying any bounds other than outlives-bounds. The former has already been implemented, the latter not yet as we don't imply any outlives-bounds at all at the moment.

Opening this issue to properly keep track of this implementation step and to clarify who is the person responsible for the implementation (me as it stands).

### PR DESCRIPTION
**Pull request by fmease**
Fixes #118479.

r? types

### CODE REVIEW

File: tests/ui/associated-type-bounds/duplicate.stderr
Comment (fmease) [inline tests/ui/associated-type-bounds/duplicate.stderr:1]: Reordering of diagnostics prolly because some queries now get called earlier than later.

**Review by compiler-errors — APPROVED** [review]

**Comment by compiler-errors** [comment]
r? compiler-errors @bors r+

**Comment by fmease** [comment]
Ah, if we want to “[treat] lazy type aliases like structs”, we should probably also imply implied bounds and make the following code compile:

```rs
#![feature(lazy_type_alias)]
type Ref0<'a, T> = Ref1<'a, T>;
type Ref1<'a, T> = &'a T;
```

@bors r-
@rustbot author

**Comment by fmease** [comment]
I accidentally squashed the commits ^^', I can try to untangle it again if you'd like me to  >.<

**Comment by compiler-errors** [comment]
No it's fine

**Review by compiler-errors — APPROVED** [review]

**Pull request merged**